site/content/troubleshooting-guide.md

```diff
@@ -1 +1,129 @@
-### TODO
\ No newline at end of file
+
+
+
+
+
+# Troubleshooting Guide
+
+This guide outlines some failure scenarios and their possible solutions.
+
+## Node.js not installed
+AWS.Deploy.Tools relies on AWS Cloud Development Kit (CDK) to provision resources for your cloud application. AWS CDK requires Node.js to be installed in your machine.
+
+**Minimum Node.js version >= 10.13.0**
+
+See [here](https://nodejs.org/en/download/) to install Node.js on your system.
+
+## Docker related issues
+AWS.Deploy.Tool requires Docker to perform containerized deployments.
+
+### Docker not installed
+See [here](https://docs.docker.com/get-docker/) to install Docker for your operating system.
+
+### Docker not running in Linux mode
+If you are on a Windows operating system, it is likely that you are running Docker in a Windows container. AWS.Deploy.Tools requires Docker to be running in Linux container mode.
+
+See [here](https://docs.docker.com/desktop/windows/#switch-between-windows-and-linux-containers) to switch between Windows and Linux containers.
+
+### Failed to build Docker Image
+Sometimes the Docker build command may fail due to the following reasons.
+
+#### Invalid Docker execution directory
+The Docker execution directory is the working directory for the Docker build command and this is where all relative paths in the docker file are resolved from. By default, the execution directory is set to the project solution directory. 
+
+However, it is possible that the Docker file is referencing projects outside the solution directory. In that case you must specify a different execution directory that can correctly evaluate project dependencies.
```

Comment:
nit throughout this entry: My impression is that "Dockerfile" is more conventional over "Docker file"

Reply:
Replaced all occurrences of _"Docker file" -> "Dockerfile"_